src/NUnitFramework/framework/Internal/Builders/CombinatorialStrategy.cs

```diff
@@ -40,7 +40,16 @@ public class CombinatorialStrategy : ICombiningStrategy
         /// <returns>The test cases.</returns>
         public IEnumerable<ITestCaseData> GetTestCases(IEnumerable[] sources)
         {
-            List<ITestCaseData> testCases = new List<ITestCaseData>();
+            return GetCombinations<ITestCaseData>(sources, (o) => new TestCaseParameters(o));
+        }
+
+        /// <summary>
+        /// Returns an <see cref="IEnumerable{T}"/> representing every single possible combination of the given parameters
+        /// </summary>
+        /// <returns>Every combination of parameters</returns>
+        public static IEnumerable<T> GetCombinations<T>(IEnumerable[] sources, Func<object[], T> handle)
```

Comment:
`handle` would be clearer to me if it was called `resultSelector` or something similar.

Reply:
I personally do not have any strong opinions on this, if you want ```resultSelector``` then that is fine by me.

Reply:
@jerryk414 What do you think of the name `IEnumerable[] valuesByParameter`?

Reply:
I don't see the point of introducing a static method here. Like all statics, it ain't pretty, and it doesn't seem to do anything different from `new CombinatorialStrategy().GetTestCases()`. Am I missing something?

I see a great advantage as well in sticking with the use of the existing internal interface. TestCaseData could have an internal method that takes an `ICombiningStrategy`, which would be used in turn by the public `Combinatorial` method.

I also dislike changing the internal structure of just one of our combining strategies, which I believe currently maintain the same structure.